common/src/main/java/org/apache/rocketmq/common/protocol/header/namesrv/BrokerHeartbeatRequestHeader.java

```diff
@@ -28,7 +28,10 @@ public class BrokerHeartbeatRequestHeader implements CommandCustomHeader {
     private String brokerAddr;
     @CFNotNull
     private String brokerName;
-
+    @CFNotNull
+    private int epoch;
+    @CFNotNull
+    private long maxOffset;
```

Comment:
IMO,  It would be better to add confirmOffset to the heartbeat packet.

Reply:
Yep~ Done!